src/index.ts

```diff
@@ -147,31 +186,74 @@ export class CSSStyleObserver {
     targetElement.style.removeProperty('transition-behavior');
   }
 
+  /**
+   * Process observed variables and collect changes
+   *
+   * @param computedStyle The computed style of the target element
+   * @param processChange Function to process each changed property
+   * @returns Collected changes
+   */
+  private _processObservedVariables<T>(
+    computedStyle: CSSStyleDeclaration,
+    processChange: (propertyName: string, currentValue: string, previousValue: string, hasChanged: boolean) => T
+  ): T[] {
+    const changes: T[] = [];
+
+    this._observedVariables.forEach((propertyName) => {
+      const currentValue = computedStyle.getPropertyValue(propertyName);
+      const previousValue = this._cachedValues[propertyName] || '';
+      const hasChanged = currentValue !== previousValue;
+
+      if (this._notificationMode === NotificationMode.ALL || hasChanged) {
+        changes.push(processChange(propertyName, currentValue, previousValue, hasChanged));
```

Comment:
Can you refactor this so that it does not call `processChange` here, but that it only stores the change object?

Reply:
refactored as requested